frontends/p4/commonInlining.h

```diff
@@ -157,7 +157,7 @@ class AbstractInliner : public Transform {
 };
 
 template <class InlineList, class InlineWorkList>
-class InlineDriver : public Transform {
+class InlineDriver : public Visitor {
```

Comment:
This suggests that the driver does not change anything, which is not really true.

Reply:
Visitor is the base class of all visitors, so encapsulates both visitors that change things and visitors that do not.  Deriving directly from Visitor means it does not use the normal Inspector/Modifier/Transform depth-first traversal.

Reply:
All of our Visitors *except* Inspectors (potentially) mutate the program.  Other direct Visitor subclasses like PassManager and PassRepeated may modify the program.  I guess we could add a comment to all such passes that they might modify the program, but it seems obvious from them not being Inspectors.